components/automate-deployment/pkg/converge/converger.go

```diff
@@ -127,63 +136,460 @@ func StartConverger(opts ...ConvergerOpt) Converger {
 	return converger
 }
 
+func (converger *converger) run() {
+	for {
+		select {
+		case d := <-converger.debug:
+			d.Do(converger)
+		default:
+		}
```

Comment:
The idea here was to not starve debug messages. While we are only using this in the test currently, I thought it might be good to eventually have grpcCalls that let us use it in production.